File: Data Structures & Algorithms/search-2d-matrix/submission-1.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        # since the entire matrix is sorted treat is as a single array or a 1d array
        # get the total number of element by row * column 
        # row
        m = len(matrix)
        # column
        n = len(matrix[0])
        l=0
        r=m*n-1
        while l<=r:
            mid = l+(r-l)//2
            # index exist in which row
            row = mid//n
            # which position inside the row
            col = mid%n
            if matrix[row][col] == target:
                return True
            elif matrix[row][col] > target:
                r=mid-1
            else:
                l=mid+1
        return False

File: Data Structures & Algorithms/merge-triplets-to-form-target/submission-1.js
class Solution {
    /**
     * @param {number[][]} triplets
     * @param {number[]} target
     * @return {boolean}
     */
    mergeTriplets(triplets, target) {
        let x = false,
            y = false,
            z = false;

        for(const [a, b, c] of triplets){
            x|= a === target[0] && b <= target[1] && c <= target[2];
            y|= a <= target[0] && b === target[1] && c <= target[2];
            z|= a <= target[0] && b <= target[1] && c === target[2];
            if( x && y && z) return true;
        }
        return false;
    }
}
